consensus/hotstuff/voteaggregator/vote_collectors.go

```diff
@@ -22,15 +35,14 @@ func (v *voteCollectors) GetOrCreateCollector(view uint64, blockID flow.Identifi
 
 // PruneUpToView prunes all collectors which target same view
 func (v *voteCollectors) PruneUpToView(view uint64) error {
-	v.lock.Lock()
-	defer v.lock.Unlock()
-	for blockID := range v.blockIDsByView[view] {
-		delete(v.collectors, blockID)
-	}
-	delete(v.blockIDsByView, view)
-	return nil
+	panic("implement me")
```

Comment:
All good except this one, pruning logic is good, we shouldn't remove it.

Reply:
Wouldn't we [implement a similar logic like this](https://github.com/onflow/flow-go/blob/feature/consensus-voting/module/forest/leveled_forrest.go#L67-L92)?

Reply:
Wasn't expecting that we will skip views, but sure, more general approach works as well